rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -595,7 +595,7 @@ class ImplItemNode extends ImplOrTraitItemNode instanceof Impl {
     exists(TraitItemNode trait |
       trait = this.resolveTraitTy() and
       trait.hasCanonicalPath(c2) and
-      if this.hasCanonicalPath(c2)
+      if trait.hasCanonicalPath(c1)
```

Comment:
I figured out what happens; this needs to be reverted, and then we need to add `c = this and` before the line `this.getName() = "core" and` inside `CrateItemNode.providesCanonicalPathPrefixFor`.